src/App.tsx

```diff
@@ -1,25 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const getPageItems = (
+  total: number,
+  itemsPerPage: number,
+  currentPage: number,
+) => {
+  return [...getNumbers(1, total)].slice(
+    itemsPerPage * (currentPage - 1),
+    itemsPerPage * currentPage,
+  );
+};
+
+const ITEMS_COUNT = 42;
+const PER_PAGE_COUNT = 5;
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(PER_PAGE_COUNT);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const pageItems = getPageItems(ITEMS_COUNT, perPage, currentPage);
+
+  const handlePerPageSelectorChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
```

Comment:
When changing the items per page, you're resetting the current page to 1. This is a good practice to avoid viewing an empty page when the previous current page number exceeds the new total page count. However, it's important to ensure that this behavior is intended and aligns with the user experience requirements.